crates/bevy_ecs/src/system/query.rs

```diff
@@ -54,9 +54,18 @@ impl<'a, Q: HecsQuery> Query<'a, Q> {
             {
                 // SAFE: we have already checked that the entity/component matches our archetype access. and systems are scheduled to run with safe archetype access
                 unsafe {
-                    self.world
-                        .get_ref_at_location_unchecked(location)
-                        .map_err(QueryError::ComponentError)
+                    if Q::Fetch::get(
+                        &self.world.archetypes[location.archetype as usize],
+                        location.index,
+                    )
+                    .map_or(true, |f| f.should_skip())
+                    {
+                        Err(QueryError::ComponentError(ComponentError::NoSuchEntity))
+                    } else {
+                        self.world
+                            .get_ref_at_location_unchecked(location)
+                            .map_err(QueryError::ComponentError)
+                    }
```

Comment:
I am clearly not confident on this part of PR and would gladly have a thorough review here.

Reply:
Hmm this has significant performance implicantions because we are now doing a "full lookup" of every component storage in the query. I'm not sure thats worth the tradeoff when we can use "Query One" when this behavior is required.

I'm down to merge the "query one" part of this PR now, but this change will require some serious thought / im pretty sure its not worth it.

Reply:
Ok, I see.
This is effectively not worth it.
In my case I have created all the camera I need in a startup components.

Reply:
Ok, I see and will update the PR to only include the query_one part.